### ISSUE TITLE
Suggestion: allow single extension in ng-accept

### ISSUE THREAD

**Issue by redders6600** [danialfarid/angular-file-upload#573]
With Chrome:

- using `ng-accept="'*.apk'" will allow selecting `.apk` files but the file picker doesn't filter anything
- using `ng-accept="'.apk'" filters the file picker, but rejects `.apk` files after it's picked.
- using `ng-accept="'.apk,"` (note comma) will allow selecting `.apk` files with a filtered file picker dialog, this is what `ng-accept="'.apk'"` _should_ do.

I believe this is because the `globStringToRegex` function is used to filter out files after they're accepted. Given `.apk` isn't a glob, `globStringToRegex` converts it into a regex like `/^\.apk$/`, which obviously doesn't match apk files with a filename.

I suggest adding another line to the `globStringToRegex` function, which (if no comma is present) checks for the first character being a `.`, and creates a wildcarded glob like so:

```javascript
function globStringToRegex(str) {
	if (str.length > 2 && str[0] === '/' && str[str.length -1] === '/') {
		return str.substring(1, str.length - 1);
	}
	var split = str.split(','), result = '';
	if (split.length > 1) {
		for (var i = 0; i < split.length; i++) {
			if (split[i].indexOf('.') == 0) {
				split[i] = '*' + split[i];
			}
			result += '(' + globStringToRegex(split[i]) + ')';
			if (i < split.length - 1) {
				result += '|'
			}
		}
    } else if (str[0] === '.') {  // .ext
      result += globStringToRegex( '*' + str ); 
	} else {
		result = '^' + str.replace(new RegExp('[.\\\\+*?\\[\\^\\]$(){}=!<>|:\\' + '-]', 'g'), '\\$&') + '$';
		result = result.replace(/\\\*/g, '.*').replace(/\\\?/g, '.');
	}
	return result;
}
```

This means `globStringToRegex('.apk') === "^\.apk$"` which is much more useful.

**Comment by danialfarid** [comment]
fixed in version 3.0.5

**Comment by redders6600** [comment]
Speedy!
